ASIM/dev/ASimTester/filteringTest/ASimFilteringTest.py

```diff
@@ -307,81 +307,155 @@ def get_values_for_dynamic_tests(self, rows):
                     break
         return values
         
-    
+
     # Performing assertions for dynamic tests with parameter filtering. Values for the parameter are taken from values_list
-    def dynamic_tests_assertions(self, param_name, query_definition, column_name_in_table, values_list, no_filter_response):
-        pass #TODO will be added in next PR
+    def dynamic_tests_assertions(self, param_name, query_definition, column_name_in_table, values_list, no_filter_rows):
+        val_str = create_values_string(values_list)
+        query_with_filter = query_definition + create_execution_strings_with_one_parameter(param_name,f"dynamic([{val_str}])" ,column_name_in_table)
+        filtered_response = self.send_query(query_with_filter)
+        filtered_rows = filtered_response.tables[0].rows
+        with self.subTest():
+            self.assertNotEqual(0, len(filtered_rows), f"Parameter: {param_name} - Got no results at all after filtering. Filtered by value: {val_str}")  
+        with self.subTest():
+            self.assertLess(len(filtered_rows), len(no_filter_rows),  f"Parameter: {param_name} - Expected to have less results after filtering. Filtered by value: {val_str}")
 
 
     # Performing filtering with one and two values (if possible) for dynamic parameters.
-    def dynamic_tests_helper(self, param_name, query_definition, no_filter_response, column_name_in_table, values_list, test_type):
-        pass #TODO will be added in next PR
-    
-    # Performing filtering for dynamic parameters with full values from no_filter_response (similar test will be done for substrings/prefixes)
-    def dynamic_full_values_tests(self, param_name, query_definition, no_filter_response, column_name_in_table):
-        pass #TODO will be added in next PR
+    def dynamic_tests_helper(self, param_name, query_definition, no_filter_rows, column_name_in_table, values_list, test_type):
+        if len(values_list) == 0:
+            self.fail(f"Parameter: {param_name} - Unable to find substrings to perform {test_type} tests")
+        # Performing filtering with one value in the parameter
+        self.dynamic_tests_assertions(param_name, query_definition, column_name_in_table, [values_list[0]], no_filter_rows )
+
+        # Performing filtering with two values in the parameter if possible
+        if len(values_list) == 1:
+            self.fail("Parameter: {param_name} - Not enough data to perform two values {test_type} tests")
+        else:
+            self.dynamic_tests_assertions(param_name,query_definition, column_name_in_table, values_list, no_filter_rows)
 
+    
+    # Performing filtering for dynamic parameters with values from no_filter_rows. "full values" because similar test are done for substrings/prefixes of values.
+    def dynamic_full_values_tests(self, param_name, query_definition, no_filter_rows, column_name_in_table):
```

Comment:
nit:
try to give a more informative name as the word "full" is misleading

Reply:
Changed the name of the function to "default" tests because the test in this function is common to all parameters of dynamic type.